configuration/src/main/java/io/airlift/configuration/ConfigurationLoader.java

```diff
@@ -57,9 +60,10 @@ public static Map<String, String> loadPropertiesFrom(String path)
             throws IOException
     {
         Properties properties = new Properties();
-        try (InputStream inputStream = new FileInputStream(path)) {
-            properties.load(inputStream);
-        }
+        String escapedProperties = readAllLines(Path.of(path), UTF_8).stream()
+                .map(line -> line.replaceAll("\\\\,", "\\\\\\\\,"))
+                .collect(joining("\n"));
+        properties.load(new StringReader(escapedProperties));
```

Comment:
What's the purpose of this?

As an aside, this is a behavioral change. By default, properties files are expected to be encoded using ISO 8859-1. I think it makes sense to change to UTF-8 nowadays, but this should be done separately and intentionally.